boards: renesas: Add support Renesas ek_ra8d2 board

khoa-nguyen-18 opened the pull request:
- Add support Renesas `RA8D2` SoC
- Add support Renesas `ek_ra8d2_r7ka8d2kflcac_cm85` board
- Add support `arducam_cu450_ov5640` camera shield, `rtklcdpar1s00001be` display shield for `ek_ra8d2_r7ka8d2kflcac_cm85`
- Add support tests and samples app to verify `ek_ra8d2_r7ka8d2kflcac_cm85` peripherals

khoatranyj committed:
manifest: update rev of hal_renesas to latest

Update rev of hal_renesas to latest

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
soc: renesas: ra: Add support Renesas RA8D2 SoC

Add support Renesas RA8D2 SoC

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>
Signed-off-by: Khoa Nguyen <khoa.nguyen.xh@renesas.com>

khoatranyj committed:
dts: arm: renesas: ra: Add support Renesas r7ka8d2kflcac SoC

- Add support Renesas r7ka8d2kflcac SoC.
- Move sdram-controller node from r7ka8p1kflcac.dtsi to ra8x2.dtsi
since this device node is available for all RA8x2 SoCs

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
boards: renesas: Add support Renesas ek_ra8d2 board

Add support Renesas ek_ra8d2 board

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>
Signed-off-by: Khoa Nguyen <khoa.nguyen.xh@renesas.com>

khoatranyj committed:
tests: drivers: i2c: Add support for i2c_api on ek_ra8d2

Add support for test app "i2c_api" on Renesas ek_ra8d2

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
tests: drivers: uart: Add support for uart_async_api on ek_ra8d2

Add support for test app "uart_async_api" on Renesas ek_ra8d2

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
tests: drivers: spi: Add support for spi_loopback on ek_ra8d2

Add support for test app "spi_loopback" on Renesas ek_ra8d2

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
tests: drivers: pwm: Add pwm tests support for ek_ra8d2

Add support for test apps on Renesas ek_ra8d2:
- tests/drivers/pwm/pwm_api
- tests/drivers/pwm/pwm_loopback

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
tests: drivers: comparator: Add support gpio_loopback on ek_ra8d2

Add support for test app "gpio_loopback" on Renesas ek_ra8d2

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
tests: subsys: pm: Add support for power_mgmt_soc on ek_ra8d2

Add support for test app power_mgmt_soc on Renesas ek_ra8d2

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
samples: drivers: counter: Add support for alarm on ek_ra8d2

Add support for sample app alarm on Renesas ek_ra8d2

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
tests: drivers: counter: Add test support for Renesas ek_ra8d2

Add support for test app on Renesas ek_ra8d2:
- tests/drivers/counter/counter_basic_api

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
tests: drivers: i2s: Add tests support for Renesas ek_ra8d2

Add support for i2s tests app on Renesas RA ek_ra8d2:
- tests/drivers/i2s/i2s_api
- tests/drivers/i2s/i2s_speed

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
samples: drivers: i2s: Add sample support for Renesas ek_ra8d2

Add support for i2s sample app on Renesas RA ek_ra8d2:
- samples/drivers/i2s/output

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
tests: drivers: dma: Add test support dma driver on ek_ra8d2

This commit adds testing support for dma driver on ek_ra8d2:
- tests/drivers/dma/chan_blen_transfer
- tests/drivers/dma/loop_transfer

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
boards: shields: Add support rtklcdpar1s00001be for ek_ra8d2

Add support rtklcdpar1s00001be display shield for Renesas RA
ek_ra8d2 board

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
tests: drivers: display: Add support display_read_write on ek_ra8d2

Add support test app display_read_write for
Renesas RA ek_ra8d2 board

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
tests: drivers: sdhc: Add tests support for Renesas ek_ra8d2 board

Add tests support for Renesas ek_ra8d2 boards

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
tests: subsys: sd: Add tests support for Renesas ek_ra8d2 board

Add tests support for Renesas ek_ra8d2 boards:
- tests/subsys/sd/sdmmc

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
tests: drivers: disk: Add tests support for Renesas ek_ra8d2 board

Add tests support for Renesas ek_ra8d2 boards:
- tests/drivers/disk/disk_access
- tests/drivers/disk/disk_performance

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
tests: subsys: fs: Add tests support for Renesas ek_ra8d2 board

Add tests support for Renesas ek_ra8d2 boards:
- tests/subsys/fs/ext2
- tests/subsys/fs/fat_fs_api

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
samples: subsys: fs: Add tests support for Renesas ek_ra8d2 board

Add tests support for Renesas ek_ra8d2 boards:
- samples/subsys/fs

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
samples: modules: lvgl: demos: add support for ek_ra8d2

Add support for ek_ra8d2/r7ka8d2kflcac/cm85 running on
rtklcdpar1s00001be display shield

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
boards: shields: Add support ArduCam CU450 OV5640 for ek_ra8d2

Add support ArduCam CU450 OV5640 shield for Renesas ek_ra8d2 board

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

thenguyenyf commented on boards/renesas/ek_ra8d2/ek_ra8d2.dtsi:
```diff
@@ -0,0 +1,224 @@
+			   <2 0 &ioport1 3 0>,	/* CS   */
+			   <3 0 &ioport1 2 0>,	/* SCK  */
+			   <4 0 &ioport1 1 0>,	/* MISO */
+			   <5 0 &ioport1 0 0>,	/* MOSI */
```
Wrong information, swap `&ioport1 1` to MOSI and `&ioport1 0` to MISO
<img width="925" height="52" alt="image" src="https://github.com/user-attachments/assets/be8f20aa-5811-4d53-a4df-072cdaa3d473" />

thaoluonguw commented on dts/arm/renesas/ra/ra8/r7ka8d2xf.dtsi:
```diff
@@ -0,0 +1,802 @@
+			 "port-irq24",
+			 "port-irq25";
+	port-irq0-pins = <6>;
+	port-irq2-pins = <13>;
```
Lack of IRQ1 in pin 5, IRQ2 lack of pin 3

thaoluonguw commented on dts/arm/renesas/ra/ra8/r7ka8d2xf.dtsi:
```diff
@@ -0,0 +1,802 @@
+			 "port-irq25";
+	port-irq0-pins = <6>;
+	port-irq2-pins = <13>;
+	port-irq3-pins = <8 12>;
```
Lack of pin 2 in irq3

khoa-nguyen-18 replied:
Thank you. We've updated it also

thaoluonguw commented on dts/arm/renesas/ra/ra8/r7ka8d2xf.dtsi:
```diff
@@ -0,0 +1,802 @@
+	port-irq21-pins = <14>;
+	port-irq23-pins = <11>;
+	port-irq24-pins = <10>;
+	port-irq25-pins = <7 9>;
```
Lack of irq26 in pin 4

thaoluonguw commented on dts/arm/renesas/ra/ra8/r7ka8d2xf.dtsi:
```diff
@@ -0,0 +1,802 @@
+	port-irq6-pins = <1>;
+	port-irq8-pins = <5>;
+	port-irq9-pins = <4>;
+	port-irq23-pins = <11>;
```
Lack of irq22 in pin 12

khoa-nguyen-18 replied:
Thank you. We've updated it also

thaoluonguw commented on dts/arm/renesas/ra/ra8/r7ka8d2xf.dtsi:
```diff
@@ -0,0 +1,802 @@
+	port-irq24-pins = <10>;
+	port-irq25-pins = <9>;
+	port-irq26-pins = <8>;
+	port-irq27-pins = <7>;
```
Lack of  pin 13 in irq27

khoatranyj committed:
dts: arm: renesas: ra: Correct number of port pins for RA8x2 series

Fix the GPIO port "ngpios" for RA8x2 series SoCs to match
the values specified in the Hardware User’s Manual (HUM).

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoa-nguyen-18 commented on dts/arm/renesas/ra/ra8/r7ka8d2xf.dtsi:
```diff
@@ -0,0 +1,802 @@
+	port-irq26-pins = <8>;
+	port-irq27-pins = <7>;
+	port-irq28-pins = <6>;
+	port-irq29-pins = <3>;
```
Thank you. We've updated it

khoa-nguyen-18 commented:
- Rebase to solve the conflicts
- Update based on the above feedback comments

thaoluonguw reviewed (approved):
LGTM now. Thank you.
